Gnome shell extensions: improvements

piegamesde opened the pull request:
###### Motivation for this change

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

CC @maralorn @rhoriguchi

piegamesde committed:
gnomeExtensions: expose gnome41Extensions in top-level

piegamesde committed:
gnomeExtensions: improve override mechanism

The reduced reduncancy should help us avoid making some copy-paste errors,
as happened previously. Also, increased ergonomics.

piegamesde commented on pkgs/top-level/all-packages.nix:
```diff
@@ -30882,7 +30882,9 @@ with pkgs;
     gnome38Extensions
-    gnome40Extensions;
+    gnome40Extensions
+    gnome41Extensions
```
Is there a way to simply add all attributes of the file to `all-packages.nix`? We already forgot that line in the GNOME 41 update, so it is quite probable that things will be forgotten again. The only solution I can think of would be to remove this and add `// (callPackage ../desktops/gnome/extensions {})` at the end. But somehow am not that fond of that idea.

maralorn replied:
Only alternative I can think of would be gnome.extensions41 and you probably don't like that either.

rhoriguchi replied:
You could add a line to the [README.md](https://github.com/NixOS/nixpkgs/blob/master/pkgs/desktops/gnome/extensions/README.md) which should decrease the chance of it being forgotten.

piegamesde replied:
Done.

rhoriguchi commented:
LGTM should make manuel overrides easier to not fail at copy pasting.

piegamesde committed:
gnomeExtensions: improve README

Document that the all-packages.nix needs to be updated too (this was forgotten
in the GNOME 41 update). Also fixed typos.
